agenthub/codeact_agent/codeact_agent.py

```diff
@@ -221,6 +222,11 @@ def step(self, state: State) -> Action:
                 f'\n\nENVIRONMENT REMINDER: You have {state.max_iterations - state.iteration} turns left to complete the task.'
             )
 
+        if config.agent.is_autonomous:
+            messages.append(
+                {'role': 'agent', 'content': f'Iteration {state.iteration}'}
+            )
```

Comment:
Why add this

Reply:
Without this, the agent repeatedly gives step 1 for the given task.

Reply:
Also, it gives a hint about how many steps it has tried.

Reply:
I don't understand it either. Doesn't look right to me...

Reply:
I don't think `role == 'agent'` is a supported behavior of litellm, it should be `user` message? Did you test the code?

Reply:
Litellm doesn't check that but openai does. Ran with a custom model.